### PR TITLE
Add support for setting API Keys for APIG

### PR DESCRIPTION
**Pull request by eahefnawy**
Adds support for providing API Keys for APIG as described in [Issue 1595](https://github.com/serverless/serverless/issues/1595)

**Status:** G2M from my side

### CODE REVIEW

**Comment by pmuens** [comment]
Looks good. The only thing which is missing is a documentation (but I think there's already another issue open for this correct?).

If that's the case it's GTM from my side.

**Comment by eahefnawy** [comment]
@pmuens thanks for looking into it. I'll add the docs asap 😊

File: lib/plugins/aws/deploy/compile/events/apiGateway/README.md [hunk 1]
```diff
@@ -120,5 +120,28 @@ functions:
...
+You can specify a list of API keys to be used by your service Rest API by adding an `apiKeys` array property to the `provider` object in `serverless.yaml`. You'll also need to explicitly specify which endpoints are `private` and require one of the api keys to be included in the request by adding a `private` boolean property to the `http` event object you want to set as private.
+
+Here's an example configuration for setting API keys for your service Rest API:
+
+```yaml
+service: my-service
+provider:
+    name: aws
+    apiKeys:
+            - myFirstKey
```
Comment (pmuens) [inline lib/plugins/aws/deploy/compile/events/apiGateway/README.md]: Does it need to be indented that much?

File: lib/plugins/aws/deploy/compile/events/apiGateway/README.md [hunk 2]
```diff
@@ -120,5 +120,28 @@ functions:
...
+            - ${mySecondSecretKey} # you can hide it in a serverless variable
+functions:
+    hello:
```
Comment (pmuens) [inline lib/plugins/aws/deploy/compile/events/apiGateway/README.md]: There was recently a switch from 4 level indentation to two levels. Could you update the indentation?

File: lib/plugins/aws/deploy/compile/events/apiGateway/README.md (hunk 1 above)
Comment (eahefnawy) [inline lib/plugins/aws/deploy/compile/events/apiGateway/README.md]: Nice! you got a good eye for "spaces" xD

Comment (pmuens) [inline lib/plugins/aws/deploy/compile/events/apiGateway/README.md]: :-D Sorry for being that nasty!

**Comment by pmuens** [comment]
Looks great! Thanks for the updates. GTM from my side :shipit:

**Comment by flomotlik** [comment]
GTM

**Pull request merged**